core/embed/rust/src/translations/blob.rs

```diff
@@ -211,10 +230,34 @@ impl<'a> Translations<'a> {
             let font_table = Table::new(reader)?;
             font_table.validate()?;
         }
+
+        let kernings = if payload_reader.remaining() > 0 {
+            let kerning_reader = read_u16_prefixed_block(&mut payload_reader)?;
+
+            // construct and validate kerning table
+            let kernings = Table::new(kerning_reader)?;
+            kernings.validate()?;
+
+            // Validate by parsing the kernings table
+            for (_, kern_data) in kernings.iter() {
+                let reader = InputStream::new(kern_data);
+                KerningTable::new(reader)?;
+            }
```

Comment:
Are there any validation checks we should do here?
Maybe we can check that the keys are sorted, to optimize lookups?